hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/RowDataProjection.java

```diff
@@ -61,7 +68,10 @@ public static RowDataProjection instance(LogicalType[] types, int[] positions) {
   public RowData project(RowData rowData) {
     GenericRowData genericRowData = new GenericRowData(this.fieldGetters.length);
     for (int i = 0; i < this.fieldGetters.length; i++) {
```

Comment:
Can we do not affect the normal code path for non evolution ? Something like
```java
public RowData project(RowData rowData, CastMap castMap)
```

Reply:
Faire enough. Fixed. I expanded RowDataProjection instead of `project(RowData rowData, CastMap castMap)` because it is convenient to keep the `CastMap` inside the projection